crates/ide/src/folding_ranges.rs

```diff
@@ -264,6 +270,24 @@ fn fold_range_for_where_clause(where_clause: ast::WhereClause) -> Option<TextRan
     None
 }
 
+fn fold_range_for_multiline_match_arm(match_arm: ast::MatchArm) -> Option<TextRange> {
+    match match_arm.expr()?.syntax().kind() {
+        BLOCK_EXPR | MATCH_EXPR | ARRAY_EXPR | RECORD_EXPR => {
+            return None;
+        }
+        _ => {}
+    }
+    if let Some(newline_offset) = match_arm.syntax().text().find_char('\n') {
+        let arm_range = match_arm.syntax().text_range();
+        let end = match_arm
+            .comma_token()
+            .and_then(|comma| Some(comma.text_range().start()))
+            .unwrap_or(arm_range.end());
+        return Some(TextRange::new(newline_offset + arm_range.start(), end));
```

Comment:
I believe we can also just use the expressions range then
```suggestion
        return Some(match_arm.expr()?.syntax().text_range());
```